acceptance-tests/src/test/java/tech/pegasys/pantheon/tests/acceptance/clique/CliqueMiningAcceptanceTest.java

```diff
@@ -72,4 +72,23 @@ public void shouldStallMiningWhenInsufficientValidators() throws IOException {
     minerNode1.verify(net.awaitPeerCount(0));
     minerNode1.verify(clique.noNewBlockCreated(minerNode1));
   }
+
+  @Test
+  public void shouldStillMineWhenSufficientValidators() throws IOException {
+    final PantheonNode minerNode1 = pantheon.createCliqueNode("miner1");
+    final PantheonNode minerNode2 = pantheon.createCliqueNode("miner2");
+    final PantheonNode minerNode3 = pantheon.createCliqueNode("miner3");
+    cluster.start(minerNode1, minerNode2, minerNode3);
+
+    cluster.waitUntil(wait.chainHeadHasProgressedByAtLeast(minerNode1, 1, 85));
+
+    cluster.stopNode(minerNode3);
+    cluster.verifyOnActiveNodes(net.awaitPeerCount(1));
+
+    cluster.waitOnActiveNodesUntil(wait.chainHeadHasProgressedByAtLeast(minerNode1, 1));
+    cluster.verifyOnActiveNodes(clique.proposerAtChainHeadEqual(minerNode1));
+
+    cluster.waitOnActiveNodesUntil(wait.chainHeadHasProgressedByAtLeast(minerNode1, 1));
+    cluster.verifyOnActiveNodes(clique.proposerAtChainHeadEqual(minerNode2));
```

Comment:
Similarly, you can't guarantee that minerNode2 will have proposed the next block here - the chain keeps progressing outside the control of the AT so just because the chain progressed by at least 1 doesn't mean it didn't progress by 2 or more and so the proposer may not be who you expect.

Reply:
Definitely right there, we have no control other the chain. And it could really be either node. Similarly to above the intention isn't to make the hard guarantee. Bad naming really, as the name suggests it expecting the block at chain head to have the specified proposer. But the condition is actually checking a block at some point with the expected proposer. I'll think of better naming of the method at least to make that clear.